qutebrowser/app.py

```diff
@@ -822,6 +823,7 @@ def __init__(self, args):
             Argument namespace from argparse.
         """
         self._last_focus_object = None
+        self._undos = collections.deque()
```

Comment:
I'm kinda worried about adding too much things to the Application object. Maybe the `tabutils.py` from #4955 would be a suitable place to add this (as a dedicated class) once it's merged - what do you think?

Reply:
Yes! I don't think it _should_ be here it was just convenient and didn't want to that kind of architectural design part of this PR. Sorry I haven't been following that PR (or any other really). But I think that class would be more useful for general use if it returned actual AbstractTabs which is already a public API rather than creating a data class that needs to be added to every time a new use for it comes up.
Also if that tabutils is to be used from more places I have some other suggestion for it (although there is no reason I can raise a PR tweaking it later). And I am conflicted over having a service for manipulating tabs vs having a way to access windows and then being able to reach their child tabs from them.